streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -142,6 +146,25 @@ public StreamTask(final TaskId id,
     }
 
     /**
+     * Finds the source of the given topic partition.
+     *
+     * @param partition the topic partition
+     * @return
+     */
+    private SourceNode findSource(final TopicPartition partition) {
```

Comment:
Why do we need to augment this function here? I.e. by the time the stream task is created, we should have populated the `sourceByTopics` map with the pattern matched topics already, so I'm not sure if the additional computational logic is needed? cc @bbejeck .